docs/en/routing/basic-usage.md

```diff
@@ -0,0 +1,490 @@
+
+The routing configuration API is constructed with the following properties:
+
+* `path: string`: The routing path segment to match in the URL.
+* `outlet: string`: The `outlet` id used to render widgets to the associated routing path.
+* `defaultRoute: boolean`(optional): Marks the outlet as default, the application will redirect to this route automatically if an unknown routes is detected on application load.
+* `defaultParams: { [index: string]: string }`(optional): Associated default parameters (`path` and `query`), required if the default route has required params.
+* `children: RouteConfig[]`(optional): Nested child routing configuration.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		defaultRoute: true
+	},
+	{
+		path: 'about',
+		outlet: 'about-overview',
+		children: [
+			{
+				path: '{services}',
+				outlet: 'about-services'
+			},
+			{
+				path: 'company',
+				outlet: 'about-company'
+			},
+			{
+				path: 'history',
+				outlet: 'about-history'
+			}
+		]
+	}
+];
+```
+
+This example would register the following routes and outlets:
+
+| Route             | Outlet          |
+| ----------------- | --------------- |
+| `/home`           | `home`          |
+| `/about`          | `about-overview`|
+| `/about/company`  | `about-company` |
+| `/about/history`  | `about-history` |
+| `/about/{param}`  | `about-services`|
+
+The `about-services` outlet has been registered to match any path after `/about` This is at odds with the other registered outlets, `about-company` and `about-history`, however the [matching algorithm](#route-matching-algorithm) ensures that exact match is weighted higher than a parameter match.
+
+## Using the Outlet MatchDetails
+
+For every `outlet` that is match on a route change `MatchDetails` are injected into the `Outlet` widget's `renderer` property. The `MatchDetails` object contains specific details for the matched outlet.
+
+* `queryParams: { [index: string]: string }`: The query params from the matched routing.
+* `params: { [index: string]: string }`: The path params from the matched routing.
+* `isExact(): boolean`: A function indicates if the outlet is an exact match for the path.
+* `isError(): boolean`: A function indicates if the outlet is an error match for the path.
+* `type: 'index' | 'partial' | 'error'`: The type of match for the route, either `index`, `partial` or `error`.
+* `router: RouterInterface`: The router instance which can used to create links and initiate route changes.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		children: [
+			{
+				path: 'details',
+				outlet: 'home-details'
+			}
+		]
+	}
+];
+```
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
```

Comment:
I think it would be better to talk about the different match details functionality separately rather than one large example.